samples/oneapi/dpc/mpi/sources/kmeans_distr_mpi.cpp

```diff
@@ -0,0 +1,113 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include <CL/sycl.hpp>
+#include <iomanip>
+#include <iostream>
+#include <mpi.h>
+
+#ifndef ONEDAL_DATA_PARALLEL
+#define ONEDAL_DATA_PARALLEL
+#endif
+
+#include "oneapi/dal/algo/kmeans.hpp"
+#include "oneapi/dal/io/csv.hpp"
+#include "oneapi/dal/detail/mpi/communicator.hpp"
+#include "oneapi/dal/detail/spmd_policy.hpp"
+
+#include "utils.hpp"
+
+namespace dal = oneapi::dal;
+
+template <typename Float>
+inline std::vector<dal::table> split_table_by_rows(const dal::detail::data_parallel_policy& p,
+                                              const dal::table& t,
+                                              std::int64_t split_count) {
+    ONEDAL_ASSERT(split_count > 0);
+    ONEDAL_ASSERT(split_count <= t.get_row_count());
+
+    const std::int64_t row_count = t.get_row_count();
+    const std::int64_t column_count = t.get_column_count();
+    const std::int64_t block_size_regular = row_count / split_count;
+    const std::int64_t block_size_tail = row_count % split_count;
+
+    std::vector<dal::table> result(split_count);
+
+    std::int64_t row_offset = 0;
+    for (std::int64_t i = 0; i < split_count; i++) {
+        const std::int64_t tail = std::int64_t(i + 1 == split_count) * block_size_tail;
+        const std::int64_t block_size = block_size_regular + tail;
+
+        const auto row_range = dal::range{ row_offset, row_offset + block_size };
+        const auto block = dal::row_accessor<const Float>{ t }.pull(p.get_queue(), row_range, sycl::usm::alloc::device);
+        result[i] = dal::homogen_table::wrap(block, block_size, column_count);
+        row_offset += block_size;
+    }
+
+    return result;
+}
+
+void run(dal::detail::spmd_policy<dal::detail::data_parallel_policy> &policy) {
+    const auto train_data_file_name = get_data_path("data/kmeans_dense_train_data.csv");
+    const auto initial_centroids_file_name = get_data_path("data/kmeans_dense_train_centroids.csv");
+
+    const auto x_train = dal::read<dal::table>(policy.get_local(), dal::csv::data_source{ train_data_file_name });
+    const auto initial_centroids =
+        dal::read<dal::table>(policy.get_local(), dal::csv::data_source{ initial_centroids_file_name });
+
+    const auto kmeans_desc = dal::kmeans::descriptor<>()
+                                 .set_cluster_count(20)
+                                 .set_max_iteration_count(5)
+                                 .set_accuracy_threshold(0.001);
+
+    auto comm = policy.get_communicator();
+    auto rank_id = comm.get_rank();
+    auto rank_count = comm.get_rank_count();
+
+    auto input_vec = split_table_by_rows<float>(policy.get_local(), x_train, rank_count);
+    dal::kmeans::train_input local_input { input_vec[rank_id], initial_centroids };
+
+    const auto result_train = dal::train(policy, kmeans_desc, local_input);
+ 	if(comm.get_rank() == 0) {
```

Comment:
Looks like indentation is incorrect here

Reply:
Fixed